theano/tensor/nnet/ctc.py

```diff
@@ -0,0 +1,167 @@
+from __future__ import (division, absolute_import, print_function)
+import os
+import os.path
+import theano.tensor as T
+from theano import config
+from theano import gof
+from theano.gof import local_optimizer
+from theano.tensor.opt import register_canonicalize
+from theano.tensor.opt import register_stabilize
+from theano.tensor.extra_ops import cpu_contiguous
+from theano.gradient import grad_undefined
+
+ctc_enabled = config.ctc.enabled
+
+
+class ConnectionistTemporalClassification(gof.COp, gof.OpenMPOp):
+    """
+    CTC loss function wrapper.
+
+    Notes
+    -----
+    Using the wrapper requires that Baidu's warp-ctc library is installed and the
+    configuration variables `config.ctc.enabled` and `config.ctc.root` be properly
+    set.
+
+    Parameters
+    ----------
+    compute_grad
+        If set to True, enables the computation of gradients of the CTC loss function.
+    """
+    __props__ = ('compute_grad', 'default_output',)
```

Comment:
`default_output` does not need to be in `__props__` list (it is more a class attribute, not an Op property).